Source/WebGPU/WebGPU/ExternalTexture.mm

```diff
@@ -85,6 +85,7 @@
 void ExternalTexture::setCommandEncoder(CommandEncoder& commandEncoder) const
 {
     m_commandEncoders.add(commandEncoder);
+    CommandEncoder::removeInvalidEncoders(m_commandEncoders);
```

Comment:
Would it be better to remove before adding so the removal loop has one less item to consider?

Reply:
Oh yes that is a better idea. Or maybe combine the add with the removal. As currently everywhere we add, we want to compress the HashSet

Reply:
(Still need to address this feedback)